client/my-sites/plans-features-main/hooks/use-filter-plans-for-plan-features.ts

```diff
@@ -0,0 +1,71 @@
+import {
+	isBusinessPlan,
+	isEcommercePlan,
+	isFreePlan,
+	isPersonalPlan,
+	isPremiumPlan,
+	PlanSlug,
+} from '@automattic/calypso-products';
+import { GridPlan } from 'calypso/my-sites/plan-features-2023-grid/hooks/npm-ready/use-wpcom-plans-with-intent';
+
+interface Props {
+	plans: Record< PlanSlug, GridPlan >;
+	isDisplayingPlansNeededForFeature: boolean;
+	selectedPlan?: string;
+	hideFreePlan?: boolean;
+	hidePersonalPlan?: boolean;
+	hidePremiumPlan?: boolean;
+	hideBusinessPlan?: boolean;
+	hideEcommercePlan?: boolean;
+}
+
+const usFilterPlansForPlanFeatures = ( {
```

Comment:
this can go back to former form (accepting `plans: PlanSlug[]` instead of `Record< PlanSlug, GridPlan >`). Should not be a blocker though - and also subject to refactoring in the next round.